notebooks/03-query/15-pivot-unpivot.ipynb

```diff
@@ -0,0 +1,372 @@
+    "  ~ specific language governing permissions and limitations\n",
+    "  ~ under the License.\n",
+    "  -->\n",
+    "\n",
+    "\n",
+    "This tutorial demonstrates how to work with [PIVOT and UNPIVOT SQL Operators](https://druid.apache.org/docs/latest/querying/sql#pivot). \n",
+    "\n",
+    "PIVOT and UNPIVOT are experimental features of Druid 29.0.0.\n",
+    "\n",
+    "In this tutorial you perform the following tasks:\n",
+    "\n",
+    "- "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "bbdbf6ad-ca7b-40f5-8ca3-1070f4a3ee42",
+   "metadata": {},
+   "source": [
+    "## Prerequisites\n",
+    "\n",
+    "This tutorial works with Druid 29.0.0 or later.\n",
+    "\n",
+    "#### Run with Docker\n",
+    "\n",
+    "Launch this tutorial and all prerequisites using the `all-services` profile of the Docker Compose file for Jupyter-based Druid tutorials. For more information, see the Learn Druid repository [readme](https://github.com/implydata/learn-druid).\n",
+    "   "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "5007a243-b81a-4601-8f57-5b14940abbff",
+   "metadata": {},
+   "source": [
+    "## Initialization\n",
+    "\n",
+    "The following cells set up the notebook and learning environment ready for use.\n",
+    "\n",
+    "### Set up and connect to the learning environment\n",
+    "\n",
+    "Run the next cell to set up the Druid Python client's connection to Apache Druid.\n",
+    "\n",
+    "If successful, the Druid version number will be shown in the output."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "c1ec783b-df3f-4168-9be2-cdc6ad3e33c2",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import druidapi\n",
+    "import os\n",
+    "import json\n",
+    "import time\n",
+    "from datetime import datetime, timedelta\n",
+    "\n",
+    "# get druid host from param if available\n",
+    "if 'DRUID_HOST' not in os.environ.keys():\n",
+    "    druid_host=f\"http://localhost:8888\"\n",
+    "else:\n",
+    "    druid_host=f\"http://{os.environ['DRUID_HOST']}:8888\"\n",
+    "\n",
+    "# get kafka host from param if available\n",
+    "if 'KAFKA_HOST' not in os.environ.keys():\n",
+    "   kafka_host=f\"http://localhost:9092\"\n",
+    "else:\n",
+    "    kafka_host=f\"{os.environ['KAFKA_HOST']}:9092\"\n",
+    "\n",
+    "print(f\"Opening a connection to {druid_host}.\")\n",
+    "\n",
+    "#setup Druid API clients\n",
+    "druid = druidapi.jupyter_client(druid_host)\n",
+    "display_client = druid.display\n",
+    "sql_client = druid.sql\n",
+    "status_client = druid.status\n",
+    "rest_client = druid.rest\n",
+    "status_client.version"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "645f0209-436a-47e2-a1f4-be6c43de0eac",
+   "metadata": {},
+   "source": [
+    "## Ingest sample data to pivot\n",
+    "\n",
+    "The following cell ingests the sample Wikipedia data into table \"example-pivot-unpivot\":"
```

Comment:
Once your Druid environment is up and running, ingest the sample data for this tutorial.

Run the following cell to create a table called `example-pivot-unpivot`. When completed, you'll see a description of the final table.

Monitor the ingestion task process in the Druid console.

Reply:
done.. monitoring is not necessary since we added the progress bar to the display client.